celltype_annotation/analysis/copykat-rms-exploration.Rmd

```diff
@@ -0,0 +1,433 @@
+
+
+```{r}
+# Only run copykat if the final TSV does not exist
+if (file.exists(copykat_distance_tsv)) {
+  
+  copykat_df <- readr::read_tsv(copykat_distance_tsv)
+  
+} else {
+
+  # Initialize dataframe for storing copykat results
+  copykat_df <- tibble::tibble(
+    submitter = sce$celltype, 
+    # and a binary version for later classification assessment
+    submitter_binary = ifelse(
+      stringr::str_starts(submitter, "Tumor"), 
+      "Tumor", 
+      "Normal"
+    )
+  )
+  
+  distance_parameters <- c("spearman", "pearson", "euclidan")
+  
+  copykat_results <- purrr::map(
+    distance_parameters, 
+    \(distance_parameter) {
+      
+      copykat_result <- copykat(
+        rawmat = as.matrix(counts(sce)), 
+        id.type = "E", # we have Ensembl gene ids, not the default gene symbols ("S")
+        sam.name = glue::glue("{params$library_id}-{distance_parameter}"), # unique name for each 
+        distance = distance_parameter, 
+        # attempt to turn off some outputs..
+        plot.genes = FALSE,
+        output.seg = FALSE,
+        n.cores = params$threads
+      )
+      
+      # export full result
+      output_file <- file.path(output_dir,
+                              glue::glue("{params$library}_copykat_{distance_parameter}.rds"))
+      readr::write_rds(copykat_result, output_file)
+      
+            
+      # return predictions
+      return(copykat_result$prediction$copykat.pred)
+    }
+  )
+  
+  names(copykat_results) <- glue::glue("copykat_{distance_parameters}")
+  
+  # Bind up all the columns
+  copykat_df <- dplyr::bind_cols(
+    copykat_df, 
+    copykat_results
+  )
+  
+  # finally, export TSV with all copykat predictions
+  readr::write_tsv(copykat_df, copykat_distance_tsv)
+} 
+```
+
+### Comparing results
+
+
+We'll prepare the data for comparison and begin with a barplot of tumor/normal classifications across `copyKat` parameters:
+
+```{r}
+# Remove NA submitter values before assessment
+copykat_df <- copykat_df |>
+  tidyr::drop_na(submitter)
+
+# Plot the data:
+compare_copykat_results_barplot(copykat_df) + 
+  ggtitle("Comparing distance parameters")
+```
+
+Let's also look at the binary submitter celltypes to get a sense of balance:
+
+```{r}
+table(copykat_df$submitter_binary)
+```
+
+
+Next, we'll look at tables for correspondence between submitter cell type annotations and `copyKat` inferences, as well as confusion matrix results for each comparison.
+For `copyKat`, "diploid" is normal and "aneuploid" is tumor.
+
+```{r}
+copykat_df |>
+  dplyr::select(submitter, submitter_binary, copykat = copykat_euclidean) |>
+  compare_result_tables("euclidean distance") 
+
+copykat_df |>
+  dplyr::select(submitter, submitter_binary, copykat = copykat_pearson) |>
+  compare_result_tables("pearson distance") 
+
+copykat_df |>
+  dplyr::select(submitter, submitter_binary, copykat = copykat_spearman) |>
+  compare_result_tables("spearman distance") 
+
```

Comment:
1. Can you create a single vector of the distances and `purrr::map` across that to simplify this? 
2. I think I would add some concluding statements here on how you are interpreting these results. Like a statement about the negative kappas throughout and any other metrics that stand out.

Reply:
Noting I've simplified this to use `purrr`, but concluding statements are forthcoming!